### DISCUSSION TITLE
[Designer] How to customize Toolbar

### DISCUSSION THREAD

**Discussion by becory** [microsoft/AdaptiveCards#8038] [Q&A] [answered]
Hi, I am trying to customize the adaptive card designer toolbar.
I had been to move 2 selectors from the original designer.
I want to move a "Preview mode" toggle button after the host app selector.
but the **togglePreview()** function is private in the card-designer.ts
How can I use this function or have another way to set Preview mode in the designer?

![MicrosoftTeams-image (49)](https://user-images.githubusercontent.com/13202145/201041690-1b5b081d-657f-476b-9467-68adab6a1460.png)

I think a lot of function is private in card-designer that can't be customized by the designer.

**Comment by becory** [comment]
Have any chance to removeElement from CardDesigner.toolbar?
I only found that have addElement in the toolbar.

I want to remove Copy JSON payload button.

designer.toolbar.getElementById('__copyJsonButton').isVisible = false
It seems not working.

**Reply by anna-dingler** [reply]
Can you share the code where you are customizing the toolbar?

`designer.toolbar.getElementById('__copyJsonButton').isVisible = false;` needs to be called before you attach the designer to its host element (`designer.attachTo(document.getElementById("designerRootHost"));`)

**Reply by becory** [reply]
It's works now, thanks!

**Comment by anna-dingler** [comment, marked as answer]
@becory 
You can enable/disable preview mode using `designer.designerSurface.isPreviewMode = true;` and `designer.designerSurface.isPreviewMode = false;`.

You will need to create a new `ToolbarButton` that handles the preview toggling itself, and hide the given "Preview mode" button.

Hope this helps!
